src/libraries/System.IO.Compression/src/System/IO/Compression/ZipCustomStreams.cs

```diff
@@ -366,7 +375,29 @@ async ValueTask<int> Core(Memory<byte> buffer, CancellationToken cancellationTok
         public override long Seek(long offset, SeekOrigin origin)
         {
             ThrowIfDisposed();
-            throw new NotSupportedException(SR.SeekingNotSupported);
+
+            if (!CanSeek)
+            {
+                throw new NotSupportedException(SR.SeekingNotSupported);
+            }
+
+            long newPositionInSuperStream = origin switch
+            {
+                SeekOrigin.Begin => _startInSuperStream + offset,
+                SeekOrigin.Current => _positionInSuperStream + offset,
+                SeekOrigin.End => _endInSuperStream + offset,
+                _ => throw new ArgumentOutOfRangeException(nameof(origin)),
+            };
+
+            if (newPositionInSuperStream < _startInSuperStream)
+            {
+                throw new IOException(SR.IO_SeekBeforeBegin);
+            }
+
+            _superStream.Seek(newPositionInSuperStream, SeekOrigin.Begin);
```

Comment:
Shouldn't we be honoring the return value here?  I know this *should* be the same as what we calculate, but I wonder if a misbehaving stream does some partial seek?  Perhaps in that case we just flow through that misbehavior by calculating the new offset?